src/assisted_test_infra/test_infra/controllers/minikube.py

```diff
@@ -0,0 +1,222 @@
+import logging
+import os
+import time
+
+import waiting
+
+
+class MinikubeController:
+    """Manage minikube commands / config changes via k8_client
+    Allow to modify configmap and verify changes
+    Allow to stop/start resources running on minukube plus waiters
+    for actions completed till pods deleted or created.
+    Enable metrics to check memory leaks and cpu usage - TBD
+    In order to access to data resource we:
+    _get_resources_by_kind -> this is the kind of resource
+    _get_obj_resources_by_name -> get the obj resource name
+    """
+    CONFIGMAP = "ConfigMap"
+    DEPLOYMENT = "Deployment"
+    REPLICASET = "ReplicaSet"
+    POD = "Pod"
+    SERVICE = "Service"
+
+    def __init__(self, k8_client):
+        """
+        Initialized k8_client accepted from fixture
+        :param k8_client:
+        """
+        self.k8_client = k8_client
+        # mapping for resources kind
+        self.mapping_resources = {
+            self.CONFIGMAP: self._get_resources_by_kind(self.CONFIGMAP, "v1"),
+            self.DEPLOYMENT: self._get_resources_by_kind(self.DEPLOYMENT, "apps/v1"),
+            self.REPLICASET: self._get_resources_by_kind(self.REPLICASET, "apps/v1"),
+            self.POD: self._get_resources_by_kind(self.POD, "v1"),
+            self.SERVICE: self._get_resources_by_kind(self.SERVICE, "v1")}
+
+    @property
+    def namespace(self):
+        return os.environ.get("NAMESPACE", "assisted-installer")
+
+    @property
+    def configmap_name(self):
+        return "assisted-service-config"
+
+    @property
+    def configmap_data(self):
+        config_map_store = self._get_obj_resources_by_name(self.CONFIGMAP, self.configmap_name)
+        assert len(config_map_store) == 1, "configmap kind name should be unique"
+        return config_map_store[0]
+
+    @property
+    def assisted_service_name(self):
+        return "assisted-service"
+
+    def resource_kind(self, name):
+        """Get the resource kind object by name
+        :param name: ReplicaSet, Deployment, Pod, ConfigMap, Service Pod
+        :return: Resource object
+        """
+        return self.mapping_resources.get(name)
+
+    def _get_resources_by_kind(self, kind_name, api_version):
```

Comment:
```suggestion
    def _get_resources_by_kind(self, kind_name: str, api_version: str) -> SomeObject:
```
As the general comment, try to have type annotations, it makes clear what are the types manipulated by the methods

Reply:
Ack i will change it  later during this patch ping pong . lets first see that we are aligned with the concept and function . - it will be fixed for sure.

Reply:
ack , starting to fix annotation